hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -0,0 +1,561 @@
+package hkmc2
+package codegen
+
+import mlscript.utils.*, shorthands.*
+import utils.*
+
+import hkmc2.Message.MessageContext
+
+import semantics.Elaborator.State
+
+import syntax.{Literal, Tree, ParamBind}
+import semantics.*
+import scala.annotation.tailrec
+import hkmc2.semantics.Elaborator.ctx
+
+import hkmc2.utils.SymbolSubst
```

Comment:
I like to order imports:
* Scala level imports
* Project level imports
* Module level imports

```suggestion
import scala.annotation.tailrec

import mlscript.utils.*, shorthands.*
import hkmc2.utils.*
import hkmc2.utils.SymbolSubst
import hkmc2.Message.MessageContext

import syntax.{Literal, Tree, ParamBind}
import semantics.*
import semantics.Elaborator.ctx
import semantics.Elaborator.State
```